File: aoc-2022/day-01/python/part2_sol.py
f = open("../input.txt")

text = f.read().split('\n')
f.close()
# print(text)

max1, max2, max3 = 0, 0, 0

calories = 0
for t in text:
    if t == '':
        # print(calories)
        if calories > max1:
            max3 = max2
            max2 = max1
            max1 = calories
        elif calories < max1 and calories > max2:
            max3 = max2
            max2 = calories
        elif calories < max2 and calories > max3:
            max3 = calories
        calories = 0
    else:
        calories += int(t)

# print(max1)
# print(max2)
# print(max3)
# calories = []
# calorie = 0

# for t in text:
#     if t == '':
#         calories.append(calorie)
#         calorie = 0
#     else:
#         calorie += int(t)

# print(calories)
# calories.sort()

# print(max(calories))
# print(calories[-1])
# print(calories[-2])
# print(calories[-3])

# print(calories[-1] + calories[-2] + calories[-3])



print(max1 + max2 + max3)

File: aoc-2022/day-01/rust/part2_sol.rs
use std::fs;

fn main() {
    let data = fs::read_to_string("../input.txt").expect("Couldn't read file");
    let mut calories: i32 = 0;
    let mut max1 = 0;
    let mut max2 = 0;
    let mut max3 = 0;
    for val in data.lines() {
        if val == "" {
            if calories > max1 {
                max3 = max2;
                max2 = max1;
                max1 = calories;
            } else if calories > max2 {
                max3 = max2;
                max2 = calories;
            } else if calories > max3 {
                max3 = calories;
            }
            calories = 0;
        } else {
            calories += val.parse::<i32>().expect("Couldn't convert to i32");
        }
    }

    println!("{}", max1 + max2 + max3);
}